benchmark/events/ee-emit-4-args.js

```diff
@@ -0,0 +1,20 @@
+'use strict';
+const common = require('../common.js');
+const EventEmitter = require('events').EventEmitter;
+
+const bench = common.createBenchmark(main, { n: [2e6] });
+
+function main(conf) {
+  const n = conf.n | 0;
+
+  const ee = new EventEmitter();
+
+  for (var k = 0; k < 10; k += 1)
```

Comment:
Why not `let`?

Reply:
its much slower because it creates a closure for the variable in each loop iteration

Reply:
@mathiasbynens We have a lint rule re: this so that's the main reason. I think @TimothyGu tried to change it recently and feedback from @bmeurer was that we shouldn't quite yet.

Reply:
![](https://gc.gy/獵駱.png)

Reply:
Here's the relevant PR with more conversation: https://github.com/nodejs/node/pull/15648